dev/src/path.ts

```diff
@@ -44,7 +44,7 @@ export const DEFAULT_DATABASE_ID = '(default)';
  */
 const RESOURCE_PATH_RE =
   // Note: [\s\S] matches all characters including newlines.
-  /^projects\/([^/]*)\/databases\/([^/]*)(?:\/documents\/)?([\s\S]*)$/;
+  /^projects\/([^/]*)\/databases\/([^/]*)(?:\/documents(?:\/([\s\S]+))?)?$/;
```

Comment:
This new regex still has problems. It should not match:

projects/PPP/databases/DDD/wrong
projects/PPP/databases/DDD//
projects/PPP/databases/DDD/documentsBAD
projects/PPP/databases/DDD/documents//
projects/PPP/databases/DDD/documents/ok//ok

Following the database capture is empty string.
projects/PPP/databases//DDD/documents

Reply:
Most of those did not match. but I did improve the regex to not match paths with empty path segments. Thanks for the examples!